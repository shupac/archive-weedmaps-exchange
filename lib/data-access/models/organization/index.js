import { types } from 'mobx-state-tree';
import License from 'models/license';
import { Address } from 'models/address';
import Location from 'models/location';
import Brand from 'models/brand';

const Organization = types.model('OrganizationModel', {
  contactName: types.maybeNull(types.string),
  email: types.maybeNull(types.string),
  id: types.string,
  name: types.string,
  address: types.maybe(Address),
  organizationType: types.string,
  phoneNumber: types.maybeNull(types.string),
  licenses: types.array(License),
  description: types.maybeNull(types.string),
  brands: types.maybe(types.array(Brand)),
  locations: types.maybe(types.array(Location)),
});

export type OrganizationType = {
  contactName: string,
  id: string,
  email: string,
  name: string,
  organizationType: string,
  phoneNumber: string,
};

export default Organization;
