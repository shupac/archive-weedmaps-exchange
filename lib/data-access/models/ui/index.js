import { types } from 'mobx-state-tree';

const Toast = types.model({
  title: '',
  body: '',
  id: types.identifier,
});

export default Toast;
