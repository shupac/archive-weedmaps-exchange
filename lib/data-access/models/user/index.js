import { types } from 'mobx-state-tree';
import Organization from 'models/organization';

export const WmProfile = types.model({
  username: '',
  email: '',
  slug: '',
  avatar_url: '',
});

const Preferences = types.model('Preferences', {
  locationId: types.maybeNull(types.string),
  userContext: types.string,
  brandId: types.maybeNull(types.string),
});

export const User = types.model({
  id: types.string,
  wmUserId: types.number,
  preferences: Preferences,
  email: types.string,
  organization: Organization,
});

export type UserType = {
  avatarUrl: string,
  wmUserId: string,
  username: string,
  userContext: string,
};
