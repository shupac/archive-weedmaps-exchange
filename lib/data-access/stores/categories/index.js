import { types, getEnv } from 'mobx-state-tree';
import Department, { type DepartmentType } from '../../models/departments';

const transform = departments =>
  departments.map(department => {
    const { id, name, position, iconImage, avatarImage } = department;

    return Department.create({
      id,
      name,
      position,
      iconImageUrl: iconImage ? iconImage.mediumUrl : '',
      avatarImageUrl: avatarImage ? avatarImage.mediumUrl : '',
    });
  });

const CategoryStore = types
  .model('CategoryStore', {
    departments: types.array(Department),
  })
  .actions(self => ({
    async getDepartments(regionId) {
      try {
        const departments = await getEnv(self).client.fetch(
          `/buyer/regions/${regionId}/departments?include=avatar_image,icon_image`,
        );

        const transformed = transform(departments);

        self.setDepartments(transformed);
      } catch (e) {
        console.log(e);
      }
    },
    setDepartments(departments) {
      self.departments = departments;
    },
  }));

export type CategoryStoreType = {
  getDepartments: (regionId: string) => void,
  departments: Array<DepartmentType>,
};

export default CategoryStore;
