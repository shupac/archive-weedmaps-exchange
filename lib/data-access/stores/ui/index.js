import { types } from 'mobx-state-tree';
import uniqueKey from 'lib/common/unique-key';
import Toast from '../../models/ui';

const UiStore = types
  .model('UiStore', {
    isLoading: false,
    notifications: types.array(Toast),
    modalIsOpen: false,
  })
  .actions(self => ({
    notifyToast(alert) {
      const notification = {
        id: uniqueKey(),
        body: alert.body,
        title: alert.title,
      };
      self.notifications.push(notification);
    },
    dismissToast(id) {
      const notificationToClear = self.notifications.find(
        notification => notification.id === id,
      );
      self.notifications.remove(notificationToClear);
    },
    onOpenModal() {
      self.modalIsOpen = true;
    },
    onCloseModal() {
      self.modalIsOpen = false;
    },
  }));

type NotificationsType = {
  id: string,
  body: string,
  title: string,
};

export type UIStoreType = {
  onOpenModal: () => void,
  onCloseModal: () => void,
  isLoading: boolean,
  notifications: NotificationsType[],
  modalIsOpen: boolean,
};

export default UiStore;
