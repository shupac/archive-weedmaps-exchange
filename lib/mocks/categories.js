
export const mockCategories = [
  {
    "slug":"concentrates",
    "position":1,
    "name":"Concentrates",
    "id":"81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
    "iconImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
      "id":"c77396fe-4293-4911-805c-2cd3afb18949"
    },
    "avatarImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
      "id":"82c8a744-10c6-48c9-b5a0-3a90de6d2c2f"
    }
  },
  {
    "slug":"edibles",
    "position":3,
    "name":"Edibles",
    "id":"e306a27d-5076-4074-ab7a-21f6485ad4bb",
    "iconImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
      "id":"b252a9f9-f06e-4981-80db-cf74d6c7573a"
    },
    "avatarImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
      "id":"97cc29eb-0ce4-4b55-81ae-e1899b6dc7ce"
    }
  }
];