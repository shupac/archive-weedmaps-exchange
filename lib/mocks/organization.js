export default {
  address: {
    territory: 'California',
    streetAddress: '41 Discovery',
    postalCode: '12345',
    longitude: 1,
    latitude: 2,
    country: 'USA',
    city: 'Irvine',
    id: '123456',
  },
  contactName: 'test contact',
  email: 'weedmaps@weedmaps.com',
  id: '123',
  licenses: [],
  name: 'test',
  organizationType: 'test org',
  phoneNumber: '1234566789',
}

export const mockOrgBrands = [
  {
    value: "9ffabab9-75bd-4d17-b8f6-265470243155",
    text: "BLVD Supply"
  },
  {
    value: "e98a5787-2e60-4302-b566-c6454a69a91f",
    text: "Harmony Extracts"
  },
  {
    value: "fd2b3edd-3de5-497b-af07-13d4cf02f240",
    text: "Bakked"
  },
  {
    value: "4853fd7e-8c28-4dac-84ae-9fde71d90d52",
    text: "Naked THC"
  },
  {
    value: "79577912-8b01-4194-b69d-e988a91d5113",
    text: "Natures Lab Extracts"
  },
  {
    value: "3072e38a-b3fb-4f0a-b5ef-bfe9bd6c0236",
    text: "Select"
  },
  {
    value: "6786ec8a-e3ea-4f4b-b23e-b55f3c9bff6d",
    text: "Flav"
  },
  {
    value: "16326c8a-0726-4a2b-92a4-57e5f67f97db",
    text: "West Coast Cure"
  },
  {
    value: "d60ca06a-89db-4584-99a8-1f9b9f427e42",
    text: "THClear Co"
  },
  {
    value: "d04dcd7c-933d-428e-a165-e91511371793",
    text: "Automation Test Brand CAsZyXnJvr"
  },
  {
    value: "d054b34f-428e-4583-a2a4-27f4e5aea6f3",
    text: "GfarmaLabs"
  }
]

export const mockOrg = {
  id: "2f4ddb82-3b67-4780-8ae4-7f27da36d813",
  phoneNumber: "1112223333",
  organizationType: "seller",
  name: "Andrew's Organization",
  email: "weedweed420@weed.com",
  description: "Update description",
  contactName: "Andrew Sciametta",
  licenses: [],
  brands: [
    {
      id: "9ffabab9-75bd-4d17-b8f6-265470243155",
      slug: "blvd-supply",
      shippingFee: 0,
      name: "BLVD Supply",
      minimumPurchasePrice: 0,
      description: "",
      deliveryEta: {
        etaMinUnit: "day",
        etaMin: 1,
        etaMaxUnit: "day",
        etaMax: 3
      }
    },
    {
      id: "e98a5787-2e60-4302-b566-c6454a69a91f",
      slug: "harmony-extracts",
      shippingFee: 0,
      name: "Harmony Extracts",
      minimumPurchasePrice: 0,
      description: "<p><ins><strong>Our Mission:</strong></ins></p> <p><em>Get Harmonized.</em></p> <p> </p> <p>Harmony Extracts passionately champions the cannabis and those within it. Wanting to do their part for the cannabis community, Harmony Extracts develops invariably excellent extract products in a long-time heart of the industry: Denver, Colorado. Here for everyone from the average smoker to the connoisseur market, Harmony Extracts assembles a team from the Old Guard of cannabis to bring the best quality and cleanest product to market at the most affordable prices. At Harmony Extracts, excellence is their norm.</p> <p><strong><ins>History:</ins></strong></p> <p>While Harmony Extracts officially started in August 2016, the passionate team behind the brand holds more than a decade’s worth of experience in the cannabis industry. Starting in 2003, the partners of Harmony Extracts began in cultivation before expanding into extraction and other leadership positions as the Colorado marijuana market became first medical and then recreational. As this team did everything from run cultivation centers to operate dispensaries, they soon felt a void in the marijuana market for consistent and quality extract product. The founders wanted to do things the right way, improving things from both a customer service and product standpoint and introducing a professionalism to the cannabis industry. Thus, Harmony Extracts sprung onto the scene fighting tooth and nail to meet Denver’s strict quality regulations and offer consistent, reliably excellent extractions.</p> <p>Today, Harmony Extracts lines the shelves of thirty dispensaries, where patients and recreational users alike notice their quality and recognize their brand. Harmony Extracts produces their world class products from their two state-of-the-art extraction facilities in Denver.</p> <p><strong><ins>Process:</ins></strong></p> <p>Hailing from many larger names in the industry, the extractors behind Harmony Extracts joined forces and combined the best methods to create their cannabis product. All of their flowers they grow in-house, or purchase from one or two of their exclusive vendors, to ensure their best materials remain the best. Drawing from classic strains and new, exciting genetics, Harmony Extracts uses butane and propane proprietary blend extraction to create the clearest and most potent shatter and wax. Their flagship product, their “Nectar” offers users an ultra-refined hydrocarbon extract with the potency of a distillate and the flavor of a live resin. Their proprietary method of extraction allows them to create a terpene-rich solution that crystallizes THCA, enabling them to separate the high-terpene cannabinoids from the crystals and reintroduce the plant materials at the end.</p> <p><strong><ins>Products:</ins></strong></p> <p>Harmony Extracts’ products started at “good” and have grown into “great” today.hey continually reevaluate their methods to keep delivering consistent excellence to their consumers. At the 2016 Hemp Connoisseur Championship, their four entries brought home eight different awards for Best Live Resin, Shatter and Wax. Always promising a high potency, their Nectar regularly tests at a potency of 99.99% THCA</p> <p><strong><ins>Lab Testing:</ins></strong></p> <p>Harmony Extracts exclusively tests their products with TEQ Analytical Laboratories in Aurora, Colorado, double checking their extracts for potency, terpene content, and any residual solvents.</p> <p><strong><ins>Networks:</ins></strong></p> <p>Currently, Harmony Extracts distributes their concentrate products to dispensaries throughout Colorado, found in popular dispensaries including Kind Love, Good Chemistry, Colorado Harvest Company and DANK in Denver, Boulder, and Crescent View. While they currently only work with recreational stores, they soon will offer products to medical patients. Harmony Extracts hopes to expand to more states by the start of 2018.</p>",
      deliveryEta: {
        etaMinUnit: "day",
        etaMin: 1,
        etaMaxUnit: "day",
        etaMax: 3
      },
      avatarImage: {
        id: "f8754a5a-01bd-46b3-ac72-5d2ae311ad28"
      }
    },
  ],
  address: {
    id: "4e0bb99c-0afb-4311-9cdc-8514d8f0a50c",
    territory: "CA",
    streetAddress: "1201 S Figueroa St",
    postalCode: "90015",
    longitude: -118.2688169,
    latitude: 34.0417609,
    country: "us",
    city: "Los Angeles"
  }
}