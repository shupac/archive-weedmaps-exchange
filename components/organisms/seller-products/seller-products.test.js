// @flow
import { shallow } from 'enzyme';
import { Router } from 'lib/routes';
import { ToggleSwitch } from '@ghostgroup/ui';
import SellerSettingsStore from 'lib/data-access/stores/seller-settings';
import SellerProductsStore from 'lib/data-access/stores/seller-products';
import AuthStore from 'lib/data-access/stores/auth';
import mockProductsResponse from 'lib/mocks/seller-products';
import { mockWmxUser } from 'lib/mocks/user';
import { mockCategories } from 'lib/mocks/categories';
import { mockOrg } from 'lib/mocks/organization';

import { SellerProducts } from './';

const mockUserSeller = {
  ...mockWmxUser,
  preferences: {
    ...mockWmxUser.preferences,
    userContext: 'seller',
    brandId: 'e98a5787-2e60-4302-b566-c6454a69a91f',
  },
};

function setup(props) {
  const mockStore = {
    sellerProducts: SellerProductsStore.create({
      sellerProducts: mockProductsResponse.data,
      sellerProductsTotalItems: mockProductsResponse.meta.totalEntries,
    }),
    sellerSettings: SellerSettingsStore.create({
      departments: mockCategories,
    }),
    authStore: AuthStore.create({
      wmxUser: {
        ...mockWmxUser,
        preferences: {
          ...mockWmxUser.preferences,
          userContext: 'seller',
          brandId: '9ffabab9-75bd-4d17-b8f6-265470243155',
        },
      },
      org: mockOrg,
    }),
  };

  const router = {
    query: {},
  };

  const component = (
    <SellerProducts store={mockStore} router={router} {...props} />
  );
  const wrapper = shallow(component, { disableLifecycleMethods: true });
  return { wrapper, mockStore, component };
}

describe('Seller Products Page', () => {
  it('should render the component', () => {
    const { wrapper } = setup();
    expect(wrapper.exists()).toEqual(true);
  });

  it('should fetch departments on mount', () => {
    const { wrapper, mockStore } = setup();
    const instance = wrapper.instance();
    const fetchDepartments = jest
      .spyOn(mockStore.sellerSettings, 'fetchDepartments')
      .mockReturnValue();
    const fetchProducts = jest
      .spyOn(mockStore.sellerProducts, 'fetchProducts')
      .mockReturnValue();
    instance.componentDidMount();
    expect(fetchDepartments).toHaveBeenCalled();
    expect(fetchProducts).toHaveBeenCalled();
  });

  it('should search products on route change', () => {
    const { wrapper } = setup({
      router: {
        asPath: '/foo',
        query: {},
      },
    });
    const instance = wrapper.instance();
    const searchProducts = jest
      .spyOn(instance, 'searchProducts')
      .mockReturnValue();
    instance.componentDidUpdate();
    expect(searchProducts).toHaveBeenCalled();
    expect(instance.prevRoute).toEqual('/foo');
  });

  it('should re-fetch products on brand change', done => {
    const { wrapper, mockStore } = setup();
    const instance = wrapper.instance();

    const searchProducts = jest
      .spyOn(instance, 'searchProducts')
      .mockReturnValue();
    instance.componentDidMount();
    mockStore.authStore.setUser(mockUserSeller);
    setTimeout(() => {
      expect(searchProducts).toHaveBeenCalled();
      done();
    }, 100);
  });

  it('disposes of the reaction when unmounting', () => {
    const { wrapper } = setup();
    const dispose = jest.spyOn(wrapper.instance(), 'dispose');
    wrapper.unmount();
    expect(dispose).toHaveBeenCalled();
  });

  it('should not search products if route does not change', () => {
    const { wrapper } = setup({
      router: {
        asPath: '/foo',
        query: {},
      },
    });
    const instance = wrapper.instance();
    instance.prevRoute = '/foo';
    const searchProducts = jest
      .spyOn(instance, 'searchProducts')
      .mockReturnValue();
    instance.componentDidUpdate();
    expect(searchProducts).not.toHaveBeenCalled();
  });

  it('should update router state', () => {
    const { wrapper } = setup({
      router: {
        query: {
          search: 'foo',
        },
      },
    });
    const instance = wrapper.instance();
    const pushRoute = jest.spyOn(Router, 'pushRoute').mockReturnValue();
    instance.setRouterState({ foo: 'bar' });
    expect(pushRoute).toHaveBeenCalledWith(
      'sellerProducts',
      {
        search: 'foo',
        foo: 'bar',
      },
      { shallow: true },
    );
  });

  it('should handle on sort', () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();
    instance.setState({ mounted: true });
    const setActive = jest.spyOn(instance, 'setActive').mockReturnValue();
    wrapper
      .find(ToggleSwitch)
      .first()
      .props()
      .onChange();
    expect(setActive).toHaveBeenCalled();
  });

  it('should handle set active toggle', () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();
    instance.setState({ mounted: true });
    const onSort = jest.spyOn(instance, 'onSort').mockReturnValue();
    wrapper
      .find('SortButton')
      .first()
      .simulate('click');
    expect(onSort).toHaveBeenCalled();
  });

  it('should format sort parameter on search', () => {
    const { wrapper, mockStore } = setup();
    const instance = wrapper.instance();
    jest.spyOn(instance, 'getRouterState').mockReturnValue({
      sortBy: 'foo',
      ascending: false,
    });
    const fetchProducts = jest
      .spyOn(mockStore.sellerProducts, 'fetchProducts')
      .mockReturnValue();
    instance.searchProducts();
    expect(fetchProducts).toHaveBeenCalledWith({
      active: undefined,
      categories: undefined,
      page: 1,
      pageSize: 15,
      query: undefined,
      sort: '-foo',
    });
  });

  it('should update the url state on clear all', () => {
    const { wrapper } = setup({
      router: {
        query: {},
      },
    });
    const instance = wrapper.instance();
    const pushRoute = jest.spyOn(Router, 'pushRoute').mockReturnValue();
    instance.clearAll();
    expect(pushRoute).toHaveBeenCalledWith(
      'sellerProducts',
      {},
      { shallow: true },
    );
    pushRoute.mockRestore();
  });

  it('should clear filters but keep search term', () => {
    const { wrapper } = setup({
      router: {
        query: {
          search: 'foo',
        },
      },
    });
    const instance = wrapper.instance();
    const pushRoute = jest.spyOn(Router, 'pushRoute').mockReturnValue();
    instance.clearAll();
    expect(pushRoute).toHaveBeenCalledWith(
      'sellerProducts',
      {
        search: 'foo',
      },
      { shallow: true },
    );
    pushRoute.mockRestore();
  });

  it('should render empty state', () => {
    const { wrapper, mockStore } = setup();
    const instance = wrapper.instance();
    instance.setState({ mounted: true });
    mockStore.sellerProducts.setSellerProducts([]);
    mockStore.sellerProducts.setSearchProductsTotalItems(0);
    expect(wrapper.find('EmptyState').exists()).toEqual(true);
  });

  it('should update seller product on set active', () => {
    const { wrapper, mockStore } = setup();
    const instance = wrapper.instance();
    const updateSellerProduct = jest
      .spyOn(mockStore.sellerProducts, 'updateSellerProduct')
      .mockReturnValue();
    const mockProduct = mockProductsResponse.data[0];
    instance.setActive(mockProduct, true);
    expect(updateSellerProduct).toHaveBeenCalledWith({
      ...mockProduct,
      active: true,
    });
  });

  it('should update route when paginated', () => {
    const { wrapper } = setup({
      router: {
        query: {
          search: 'foo',
        },
      },
    });
    const instance = wrapper.instance();
    const pushRoute = jest.spyOn(Router, 'pushRoute').mockReturnValue();
    instance.goToPage(2);
    expect(pushRoute).toHaveBeenCalledWith(
      'sellerProducts',
      {
        search: 'foo',
        page: 2,
      },
      { shallow: true },
    );
  });

  it('should update url state for different sort key', () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();
    const setRouterState = jest
      .spyOn(instance, 'setRouterState')
      .mockReturnValue();
    instance.onSort('foo');
    expect(setRouterState).toHaveBeenCalledWith({
      sortBy: 'foo',
      ascending: true,
    });
  });

  it('should update url state for same sort key', () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();
    jest.spyOn(instance, 'getRouterState').mockReturnValue({
      sortBy: 'foo',
      ascending: true,
    });
    const setRouterState = jest
      .spyOn(instance, 'setRouterState')
      .mockReturnValue();
    instance.onSort('foo');
    expect(setRouterState).toHaveBeenCalledWith({
      sortBy: 'foo',
      ascending: false,
    });
  });

  it('should show product count for only 1 product', () => {
    const { wrapper, mockStore } = setup();
    mockStore.sellerProducts.setSellerProducts([mockProductsResponse.data[0]]);
    mockStore.sellerProducts.setSearchProductsTotalItems(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show pagination', () => {
    const { wrapper } = setup({
      router: {
        query: {
          page_size: 1,
          page: 1,
        },
      },
    });
    const instance = wrapper.instance();
    instance.setState({ mounted: true });
    expect(wrapper).toMatchSnapshot();
  });
});
