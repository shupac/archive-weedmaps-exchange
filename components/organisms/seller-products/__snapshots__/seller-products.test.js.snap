// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seller Products Page should show pagination 1`] = `
<styles__SellerProductsWrapper>
  <FilterPanel
    onClearAll={[Function]}
  >
    <withRouter(TreeFilterSection)
      defaultLabel="All Categories"
      maxHeight={500}
      paramKey="categories"
      route="sellerProducts"
      title="Categories"
      trees={
        Array [
          Object {
            "children": Array [
              Object {
                "id": "c922a456-146e-4293-955b-bb517d3063ac",
                "name": "Sauce",
              },
              Object {
                "id": "b7295924-975c-4e0c-bb18-250d4b233a35",
                "name": "Live Resin",
              },
              Object {
                "id": "2e6d3214-fac6-45a8-9786-6dbda76a6a33",
                "name": "Hash Oil",
              },
              Object {
                "id": "89af476a-0f1e-4158-9781-ac3109f3c96c",
                "name": "Solventless",
              },
              Object {
                "id": "0128b3c7-54d6-4b27-812d-718fc8ab4b4a",
                "name": "Ice Hash",
              },
              Object {
                "id": "68ea1a1c-9a5c-44a8-b97e-daceb020c284",
                "name": "Dry Sift",
              },
              Object {
                "id": "5e9573a9-a0b2-4b7e-b345-ce750ba62a41",
                "name": "Apparel",
              },
              Object {
                "id": "bfae7a36-f12f-4faf-9a63-1d4e9b60d9be",
                "name": "Rosin",
              },
              Object {
                "id": "20eef949-8169-4a80-a80e-8482a393e29d",
                "name": "Shatter",
              },
              Object {
                "id": "934e604b-7ce6-4545-a61d-c44c32072a44",
                "name": "Distillate",
              },
              Object {
                "id": "335bc5dd-15b1-4be8-8595-17cc1df3e853",
                "name": "Crumble",
              },
              Object {
                "id": "6c56279f-be48-44be-8a53-8fb8303e154c",
                "name": "Budder",
              },
            ],
            "parent": Object {
              "id": "81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
              "name": "Concentrates",
            },
          },
          Object {
            "children": Array [
              Object {
                "id": "963d5e18-0feb-4263-9bd4-2ebd1e35693b",
                "name": "Beverages",
              },
              Object {
                "id": "f245a838-fd25-4c7d-af45-e989c6702304",
                "name": "Munchies",
              },
              Object {
                "id": "bf045477-c46d-4f12-b1bf-94f0f04e7f33",
                "name": "Baked Goods",
              },
              Object {
                "id": "1e9b1182-59e6-4b9a-92ae-2c731a517409",
                "name": "Candies",
              },
              Object {
                "id": "95f27620-bc0f-421f-8d6f-d0e6abb1ec4c",
                "name": "Cooking Ingredients",
              },
              Object {
                "id": "af3eeab3-26f9-425b-99b5-fa70021e0d45",
                "name": "Gummies",
              },
              Object {
                "id": "330979e6-bacf-42e0-adbd-bf623c590a7a",
                "name": "Chocolate",
              },
            ],
            "parent": Object {
              "id": "e306a27d-5076-4074-ab7a-21f6485ad4bb",
              "name": "Edibles",
            },
          },
        ]
      }
    />
    <withRouter(FilterSection)
      defaultLabel="All Availability"
      options={
        Array [
          Object {
            "id": "published",
            "name": "Published",
          },
          Object {
            "id": "unpublished",
            "name": "Unpublished",
          },
        ]
      }
      paramKey="active"
      route="sellerProducts"
      singleSelection={true}
      title="Availability"
    />
  </FilterPanel>
  <styles__Content>
    <withRouter(inject-SearchBar-with-store)
      queryParams={
        Array [
          "categories",
          "active",
          "sortBy",
          "ascending",
          "page",
        ]
      }
      route="sellerProducts"
      showCategory={false}
    />
    <styles__ProductsTable>
      <styles__TableHeader>
        <styles__TableCell
          key="name"
        >
          <SortButton
            isActive={true}
            onClick={[Function]}
            pointsUp={true}
          >
            Name
          </SortButton>
        </styles__TableCell>
        <styles__TableCell
          key="category_names"
        >
          <SortButton
            isActive={false}
            onClick={[Function]}
            pointsUp={true}
          >
            Category
          </SortButton>
        </styles__TableCell>
        <styles__TableCell
          key="quantity"
        >
          <SortButton
            isActive={false}
            onClick={[Function]}
            pointsUp={true}
          >
            Inventory
          </SortButton>
        </styles__TableCell>
        <styles__TableCell
          key="zone_count"
        >
          <SortButton
            isActive={false}
            onClick={[Function]}
            pointsUp={true}
          >
            Zones
          </SortButton>
        </styles__TableCell>
        <styles__TableCell
          key="active"
        >
          <SortButton
            isActive={false}
            onClick={[Function]}
            pointsUp={true}
          >
            Published
          </SortButton>
        </styles__TableCell>
      </styles__TableHeader>
      <styles__TableBody>
        <styles__TableCell>
          <styles__AvatarName>
            <img
              alt="24K Live Resin Shatter"
              src="https://images-acceptance.weedmaps.com/products/000/024/998/avatar/square/natureslab_live24k-1.jpg"
            />
            <Link
              href="/seller/products/b3f4fd48-6698-43cd-addc-33504fcd0434"
            >
              <a>
                24K Live Resin Shatter
              </a>
            </Link>
          </styles__AvatarName>
        </styles__TableCell>
        <styles__TableCell>
          Concentrates &gt; Live Resin
        </styles__TableCell>
        <styles__TableCell>
          5 in stock for 2 variants
        </styles__TableCell>
        <styles__TableCell>
          <styles__NoValue>
            0
          </styles__NoValue>
        </styles__TableCell>
        <styles__TableCell>
          <ToggleSwitch
            checked={false}
            disabled={false}
            hideLabel={true}
            onChange={[Function]}
          />
        </styles__TableCell>
        <styles__TableCell>
          <styles__AvatarName>
            <img
              alt="2 Palms Tee"
              src="https://images-acceptance.weedmaps.com/products/000/016/381/avatar/square/blvdsupply_2_palmstee_1.png"
            />
            <Link
              href="/seller/products/914a63ff-a67b-47fb-946b-86c991ce4560"
            >
              <a>
                2 Palms Tee
              </a>
            </Link>
          </styles__AvatarName>
        </styles__TableCell>
        <styles__TableCell>
          Gear &gt; Men's Apparel
        </styles__TableCell>
        <styles__TableCell>
          5 in stock for 1 variant
        </styles__TableCell>
        <styles__TableCell>
          <styles__NoValue>
            0
          </styles__NoValue>
        </styles__TableCell>
        <styles__TableCell>
          <ToggleSwitch
            checked={false}
            disabled={false}
            hideLabel={true}
            onChange={[Function]}
          />
        </styles__TableCell>
        <styles__TableCell>
          <styles__AvatarName>
            <img
              alt="303 Kush Live Wax"
              src="https://images.weedmaps.com/products/000/052/107/avatar/square/1506462031-170918_303KushLW_3874.jpg?development=1"
            />
            <Link
              href="/seller/products/554cc78e-d9f3-4240-8b57-63a377c5a240"
            >
              <a>
                303 Kush Live Wax
              </a>
            </Link>
          </styles__AvatarName>
        </styles__TableCell>
        <styles__TableCell>
          Concentrates
        </styles__TableCell>
        <styles__TableCell>
          <styles__NoValue>
            0
          </styles__NoValue>
        </styles__TableCell>
        <styles__TableCell>
          <styles__NoValue>
            0
          </styles__NoValue>
        </styles__TableCell>
        <styles__TableCell>
          <ToggleSwitch
            checked={true}
            disabled={true}
            hideLabel={true}
            onChange={[Function]}
          />
        </styles__TableCell>
      </styles__TableBody>
      <styles__TableFooter>
        Showing 
        1-3
         of 
        1841
         Products
        <PagingControls
          currentPage={1}
          onSelectPage={[Function]}
          pageCount={1841}
        />
      </styles__TableFooter>
    </styles__ProductsTable>
  </styles__Content>
</styles__SellerProductsWrapper>
`;

exports[`Seller Products Page should show product count for only 1 product 1`] = `
<styles__SellerProductsWrapper>
  <FilterPanel
    onClearAll={[Function]}
  >
    <withRouter(TreeFilterSection)
      defaultLabel="All Categories"
      maxHeight={500}
      paramKey="categories"
      route="sellerProducts"
      title="Categories"
      trees={
        Array [
          Object {
            "children": Array [
              Object {
                "id": "c922a456-146e-4293-955b-bb517d3063ac",
                "name": "Sauce",
              },
              Object {
                "id": "b7295924-975c-4e0c-bb18-250d4b233a35",
                "name": "Live Resin",
              },
              Object {
                "id": "2e6d3214-fac6-45a8-9786-6dbda76a6a33",
                "name": "Hash Oil",
              },
              Object {
                "id": "89af476a-0f1e-4158-9781-ac3109f3c96c",
                "name": "Solventless",
              },
              Object {
                "id": "0128b3c7-54d6-4b27-812d-718fc8ab4b4a",
                "name": "Ice Hash",
              },
              Object {
                "id": "68ea1a1c-9a5c-44a8-b97e-daceb020c284",
                "name": "Dry Sift",
              },
              Object {
                "id": "5e9573a9-a0b2-4b7e-b345-ce750ba62a41",
                "name": "Apparel",
              },
              Object {
                "id": "bfae7a36-f12f-4faf-9a63-1d4e9b60d9be",
                "name": "Rosin",
              },
              Object {
                "id": "20eef949-8169-4a80-a80e-8482a393e29d",
                "name": "Shatter",
              },
              Object {
                "id": "934e604b-7ce6-4545-a61d-c44c32072a44",
                "name": "Distillate",
              },
              Object {
                "id": "335bc5dd-15b1-4be8-8595-17cc1df3e853",
                "name": "Crumble",
              },
              Object {
                "id": "6c56279f-be48-44be-8a53-8fb8303e154c",
                "name": "Budder",
              },
            ],
            "parent": Object {
              "id": "81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
              "name": "Concentrates",
            },
          },
          Object {
            "children": Array [
              Object {
                "id": "963d5e18-0feb-4263-9bd4-2ebd1e35693b",
                "name": "Beverages",
              },
              Object {
                "id": "f245a838-fd25-4c7d-af45-e989c6702304",
                "name": "Munchies",
              },
              Object {
                "id": "bf045477-c46d-4f12-b1bf-94f0f04e7f33",
                "name": "Baked Goods",
              },
              Object {
                "id": "1e9b1182-59e6-4b9a-92ae-2c731a517409",
                "name": "Candies",
              },
              Object {
                "id": "95f27620-bc0f-421f-8d6f-d0e6abb1ec4c",
                "name": "Cooking Ingredients",
              },
              Object {
                "id": "af3eeab3-26f9-425b-99b5-fa70021e0d45",
                "name": "Gummies",
              },
              Object {
                "id": "330979e6-bacf-42e0-adbd-bf623c590a7a",
                "name": "Chocolate",
              },
            ],
            "parent": Object {
              "id": "e306a27d-5076-4074-ab7a-21f6485ad4bb",
              "name": "Edibles",
            },
          },
        ]
      }
    />
    <withRouter(FilterSection)
      defaultLabel="All Availability"
      options={
        Array [
          Object {
            "id": "published",
            "name": "Published",
          },
          Object {
            "id": "unpublished",
            "name": "Unpublished",
          },
        ]
      }
      paramKey="active"
      route="sellerProducts"
      singleSelection={true}
      title="Availability"
    />
  </FilterPanel>
  <styles__Content>
    <withRouter(inject-SearchBar-with-store)
      queryParams={
        Array [
          "categories",
          "active",
          "sortBy",
          "ascending",
          "page",
        ]
      }
      route="sellerProducts"
      showCategory={false}
    />
    <loader__LoaderWrapper>
      <Loader />
    </loader__LoaderWrapper>
  </styles__Content>
</styles__SellerProductsWrapper>
`;
